File: YukariLauncher/src/main/java/net/kdt/pojavlaunch/value/DependentLibrary.java
package net.kdt.pojavlaunch.value;

import androidx.annotation.Keep;

import net.kdt.pojavlaunch.JMinecraftVersionList.Arguments.ArgValue.ArgRules;

@Keep
public class DependentLibrary {
    public ArgRules[] rules;
    public String name;
    public LibraryDownloads downloads;
    public String url;

    @Keep
/**
 * LibraryDownloads内部クラス
 */
	public static class LibraryDownloads {
		public final MinecraftLibraryArtifact artifact;
/**
 * LibraryDownloadsを構築します
 */
		public LibraryDownloads(MinecraftLibraryArtifact artifact) {
			this.artifact = artifact;
		}
	}
}

File: YukariLauncher/src/main/java/com/arata/yukarilauncher/value/DependentLibrary.kt
package com.arata.yukarilauncher.value

import androidx.annotation.Keep
import com.arata.yukarilauncher.value.JMinecraftVersionList.Arguments.ArgValue.ArgRules

@Keep
class DependentLibrary {
    var rules: Array<ArgRules>? = null
    var name: String? = null
    var downloads: LibraryDownloads? = null
    var url: String? = null

    @Keep
    class LibraryDownloads(val artifact: MinecraftLibraryArtifact)
}